reactor-core/src/test/java/reactor/core/publisher/FluxBufferTimeoutTest.java

```diff
@@ -202,6 +203,53 @@ public void downstreamDemandShouldBeAbleToDecreaseOnTimeSpan() {
 		assertThat(test.scan(Scannable.Attr.REQUESTED_FROM_DOWNSTREAM)).isEqualTo(0L);
 	}
 
+	@Test
+	public void requestedFromUpstreamShouldNotExceedDownstreamDemand() {
+		Subscription[] subscriptionsHolder = new Subscription[1];
+		CoreSubscriber<List<String>> actual = new LambdaSubscriber<>(null, e -> {}, null, s -> subscriptionsHolder[0] = s);
+
+		VirtualTimeScheduler timeScheduler = VirtualTimeScheduler.getOrSet();
```

Comment:
I believe the tearDown method calls VirtualTimeScheduler::reset. It's a moot point now though, since I have changed to use VirtualTimeScheduler::create for my reworked test.